### PR TITLE
feat: [CDS-51802]: ServiceDetailPage : Env part.

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
#### Screenshots

https://user-images.githubusercontent.com/98325173/217161464-d344b468-3cfa-43b5-ba88-4b1aadd9a88a.mov

https://user-images.githubusercontent.com/98325173/217161334-60e7e26c-d418-4ba6-a198-bbfc2c069a8e.mov

image

Screenshot 2023-02-01 at 2 58 14 AM

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx [hunk 1]
```diff
@@ -463,5 +374,99 @@ export const ActiveServiceInstancePopover: React.FC<ActiveServiceInstancePopover
...
+    error: envError
+  } = useGetInstancesDetails({
+    queryParams: queryParamsEnv,
+    queryParamStringifyOptions: {
+      arrayFormat: 'repeat'
+    },
+    lazy: !isEnvDetail
+  })
+
+  if ((!isEnvDetail && loading) || (isEnvDetail && envLoading)) {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:438]: nit: comment here would be helpful

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvTable.tsx [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+                environmentType: envType,
+                infrastructureId: infraId,
+                infrastructureName: infraName,
+                instanceCount: instanceCount,
+                showEnv,
+                showEnvType,
+                showInfra
+              })
+              showEnv = false
+              showEnvType = false
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvTable.tsx:97]: What is going on here? 😂

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvTable.tsx (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ServiceDetailsSummaryV2/ServiceDetailsEnvTable.tsx:97]: These are mostly flags for whether to show a value in a column or not, in a one-to-many mapping.

File: src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/ServiceDetails/ActiveServiceInstances/ActiveServiceInstancePopover.tsx:438]: In this file, I have only exported a portion of the code, outside.
So, basically, restructuring happened here, but I will check this condition and will add a comment here.

**Review by ishantmehtaharness — APPROVED** [review]
This is reviewed earlier. approving

**Pull request merged**